x-pack/plugins/index_management/public/application/components/component_templates/component_template_list/table.tsx

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React, { FunctionComponent, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import {
+  EuiInMemoryTable,
+  EuiButton,
+  EuiInMemoryTableProps,
+  EuiTableFieldDataColumnType,
+  EuiIcon,
+} from '@elastic/eui';
+
+import { EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import { ComponentTemplateListItem } from '../types';
+
+export interface Props {
+  componentTemplates: ComponentTemplateListItem[];
+  onReloadClick: () => void;
+  onDeleteClick: (componentTemplateName: string[]) => void;
+}
+
+export const ComponentTable: FunctionComponent<Props> = ({
+  componentTemplates,
+  onReloadClick,
+  onDeleteClick,
+}) => {
+  const [selection, setSelection] = useState<ComponentTemplateListItem[]>([]);
+
+  const tableProps: EuiInMemoryTableProps<ComponentTemplateListItem> = {
+    itemId: 'name',
+    isSelectable: true,
+    'data-test-subj': 'componentTemplatesTable',
+    sorting: { sort: { field: 'name', direction: 'asc' } },
+    selection: {
+      onSelectionChange: setSelection,
+      selectable: ({ usedBy }) => usedBy.length === 0,
+      selectableMessage: (selectable) =>
+        !selectable
+          ? i18n.translate('xpack.idxMgmt.componentTemplatesList.table.disabledSelectionLabel', {
+              defaultMessage: 'Component template is in use and cannot be deleted',
+            })
+          : i18n.translate('xpack.idxMgmt.componentTemplatesList.table.selectionLabel', {
+              defaultMessage: 'Select this component template',
+            }),
+    },
+    rowProps: () => ({
+      'data-test-subj': 'componentTemplateTableRow',
+    }),
+    cellProps: (componentTemplate, column) => {
+      const { field } = column as EuiTableFieldDataColumnType<ComponentTemplateListItem>;
+
+      return {
+        'data-test-subj': `componentTemplateTableRow-${field}`,
```

Comment:
Is all the logic in `cellProps ` necessary?

Reply:
This allows the name of the cell to be part of the `data-test-subj`. I see I'm not currently using this in any of the tests though, so I removed for now.